airflow/models/dag.py

```diff
@@ -781,13 +788,16 @@ def get_task_instances(
             start_date = (timezone.utcnow() - timedelta(30)).date()
             start_date = timezone.make_aware(
                 datetime.combine(start_date, datetime.min.time()))
-        end_date = end_date or timezone.utcnow()
+
         tis = session.query(TaskInstance).filter(
             TaskInstance.dag_id == self.dag_id,
             TaskInstance.execution_date >= start_date,
-            TaskInstance.execution_date <= end_date,
             TaskInstance.task_id.in_([t.task_id for t in self.tasks]),
         )
+        # This allows ALLOW_TRIGGER_IN_FUTURE config to take affect, rather than mandating exec_date <= UTC
+        if end_date or not self.allow_future_exec_dates:
+            end_date = end_date or timezone.utcnow()
+            tis = tis.filter(TaskInstance.execution_date <= end_date)
```

Comment:
Let's add a test for this piece of code.

Reply:
@tooptoop4 , maybe add a test for this code to `test_taskinstance.py`, unless you find some better place for the test.